src/plugins/utils/js-error-utils.ts

```diff
@@ -48,7 +48,10 @@ const appendErrorPrimitiveDetails = (
     rumEvent: JSErrorEvent,
     error: any
 ): void => {
-    rumEvent.type = error.toString();
+    // Keep unhandledrejection as type as it will write to rumEvent.message
+    if (rumEvent.type !== 'unhandledrejection') {
+        rumEvent.type = error.toString();
+    }
```

Comment:
issue: We want the behavior of promise rejection events to be the same as the behavior of error events. This is so that monitoring dashboards can display errors from both sources in the same way.

Reply:
Agreed, will reflect on next revision.

Reply:
The reason for this change is due to the unit test case `'when an unhandled rejection is thrown then the plugin records the name, message'` in `JsErrorPlugin.test.ts`.

If the error's reason is simply a string, without this change the output `JsErrorEvent` becomes `{"message": "Something went wrong!", "type": "Something went wrong!", "version": "1.0.0"}`.

Even though we set `type: event.type` and `error: event.reason` in `promiseRejectEventHandler`, when we build the `JsErrorEvent` while the `event.reason` is a string, it goes through `appendErrorPrimitiveDetails` in `js-error-utils` that simply sets both the `rumEvent.type` and `rumEvent.message` with `error.toString()`.

As a result, even though the existing, expected behavior in the case where the output `JsErrorEvent` is `{"message": "Something went wrong!", "type": "unhandledrejection", "version": "1.0.0"}`, it becomes `{"message": "Something went wrong!", "type": "Something went wrong!", "version": "1.0.0"}` as mentioned above.

Reply:
It seems that the existing convention for a `PromiseRejectEvent` when its `reason` is a string is to keep the output `JsErrorEvent.type` as `'unhandledrejection'` and update `JsErrorEvent.message` as `error.toString()`

I will raise a new revision maintaining this section so we can discuss again.